cpu/stm32/include/cpu_conf.h

```diff
@@ -115,19 +115,52 @@ extern "C" {
 #define FLASHPAGE_SIZE                  (128U)
 #endif
 
+#ifdef FLASHPAGE_SIZE
 #define FLASHPAGE_NUMOF                 (STM32_FLASHSIZE / FLASHPAGE_SIZE)
+#endif
+
+#if defined(CPU_FAM_STM32F2) || defined(CPU_FAM_STM32F4) || \
+    defined(CPU_FAM_STM32F7)
+#define PERIPH_FLASHPAGE_CUSTOM_PAGESIZES
+
+/* DB1M splits the 1MB into two banks, 2MB devices are always split on stm32f4 */
+#if (defined(FLASH_OPTCR_DB1M) && (STM32_FLASHSIZE >= (1024 * 1024)))
+#define FLASHPAGE_DUAL_BANK             (1)
```

Comment:
It would be nice to have a better definition of this one. There are some assumptions here that would be better be made explicit like the fact that `DB1M` is not changed by the user, and that the current configuration is the default one.

Reply:
The other option is to detect this at runtime, but that's more complex. The other option is to extend the documentation to make this  a bit more clear

Reply:
Documentation is what I meant :)

Reply:
> better be made explicit like the fact that 

A doxygen `@note` for example :)